carbon-route.html

```diff
@@ -223,16 +223,16 @@
     __resetProperties: function() {
       this._setActive(false);
       this._matched = null;
-      //this.tail = { path: null, prefix: null, queryParams: null };
-      //this.data = {};
+      this.tail = { path: null, prefix: null, __queryParams: null };
+      this.data = {};
```

Comment:
I have removed these changes for now, but would like to propose them in a different pull request later.

The performance is obtained if `carbon-route` are mutually exclusive. However, our application breaks as this assumption is incorrect there. We have multiple steps in one of our pages which guide the user through a process. This process starts with selecting a project and then a pull request.

Given that the user has not selected a project, a message "Select a repository on the left to start" is shown. Once the user selects a project, we show a list of pull requests with the same process. Then when the user selects a pull request, we switch to a different view.

Now the issue becomes that if you stop the process and go back to the beginning (start over to project selection) all nested `carbon-route` are not notified of this change. They still think they are active as they never received the updated `tail` data. Therefore our application breaks as it is relying on corrupt and conflicting data.

Nesting of `carbon-route` is severely limited due to this behavior. While I do understand the performance improvement, I would kindly ask you to reconsider it.